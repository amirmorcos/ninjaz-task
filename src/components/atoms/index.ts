import Icon from "./Icon/Icon";
import Divider from "./Divider/Divider";
import Avatar from "./Avatar/Avatar";

export { Icon, Divider, Avatar };
