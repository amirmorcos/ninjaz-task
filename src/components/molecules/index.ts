import AppImage from "./AppImage/AppImage";
import PostItem from "./PostItem/PostItem";

export { AppImage, PostItem };
