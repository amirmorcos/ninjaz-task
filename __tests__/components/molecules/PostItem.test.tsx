import React from "react";
import renderer from "react-test-renderer";
import { PostItem } from "../../../src/components/molecules";

jest.mock("react-native-vector-icons/Ionicons", () => "Icon");
jest.mock("react-native-toast-message", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));
jest.mock("react-native-size-matters", () => ({
  scale: () => {},
  verticalScale: () => {},
  moderateScale: () => {},
}));

test("renders correctly", () => {
  const tree = renderer
    .create(
      <PostItem
        item={{
          id: "6583404d1a06525b19266460",
          image: "",
          likes: 0,
          tags: [],
          text: "Dog in a forest at sunset dog in forest with sun r...",
          publishDate: "2023-12-20T19:28:13.612Z",
          updatedDate: "2023-12-20T19:28:13.612Z",
          owner: {
            id: "60d0fe4f5311236168a10a29",
            title: "ms",
            firstName: "Vanessa",
            lastName: "Ramos",
            picture: "https://randomuser.me/api/portraits/med/women/33.jpg",
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly with tags", () => {
  const tree = renderer
    .create(
      <PostItem
        item={{
          id: "6583404d1a06525b19266460",
          image: "",
          likes: 0,
          tags: ["dog", "animal", "golden retriever"],
          text: "Dog in a forest at sunset dog in forest with sun r...",
          publishDate: "2023-12-20T19:28:13.612Z",
          updatedDate: "2023-12-20T19:28:13.612Z",
          owner: {
            id: "60d0fe4f5311236168a10a29",
            title: "ms",
            firstName: "Vanessa",
            lastName: "Ramos",
            picture: "https://randomuser.me/api/portraits/med/women/33.jpg",
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
