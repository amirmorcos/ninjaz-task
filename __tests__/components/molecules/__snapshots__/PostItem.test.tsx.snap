// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": undefined,
              "height": undefined,
              "width": undefined,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingHorizontal": undefined,
        }
      }
    >
      <Text>
        Vanessa
         
        Ramos
      </Text>
      <Text
        style={
          {
            "color": "#A9A9A9",
            "marginTop": undefined,
          }
        }
      >
        2 months ago
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": undefined,
        "paddingStart": undefined,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
    >
      Dog in a forest at sunset dog in forest with sun r...
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "borderRadius": undefined,
              "height": undefined,
              "marginVertical": undefined,
              "width": undefined,
            },
            undefined,
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        color="#B4B4B8"
        name="heart-outline"
        size={20}
      />
      <Text
        style={
          {
            "marginStart": undefined,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders  correctly with tags 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": undefined,
              "height": undefined,
              "width": undefined,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingHorizontal": undefined,
        }
      }
    >
      <Text>
        Vanessa
         
        Ramos
      </Text>
      <Text
        style={
          {
            "color": "#A9A9A9",
            "marginTop": undefined,
          }
        }
      >
        2 months ago
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": undefined,
        "paddingStart": undefined,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
    >
      Dog in a forest at sunset dog in forest with sun r...
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "borderRadius": undefined,
              "height": undefined,
              "marginVertical": undefined,
              "width": undefined,
            },
            undefined,
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
        }
      }
    >
      <Text>
        #
        dog
         
      </Text>
      <Text>
        #
        animal
         
      </Text>
      <Text>
        #
        golden retriever
         
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        color="#B4B4B8"
        name="heart-outline"
        size={20}
      />
      <Text
        style={
          {
            "marginStart": undefined,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": undefined,
              "height": undefined,
              "width": undefined,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingHorizontal": undefined,
        }
      }
    >
      <Text>
        Vanessa
         
        Ramos
      </Text>
      <Text
        style={
          {
            "color": "#A9A9A9",
            "marginTop": undefined,
          }
        }
      >
        2 months ago
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": undefined,
        "paddingStart": undefined,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
    >
      Dog in a forest at sunset dog in forest with sun r...
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "borderRadius": undefined,
              "height": undefined,
              "marginVertical": undefined,
              "width": undefined,
            },
            undefined,
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        color="#B4B4B8"
        name="heart-outline"
        size={20}
      />
      <Text
        style={
          {
            "marginStart": undefined,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with tags 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": undefined,
              "height": undefined,
              "width": undefined,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "paddingHorizontal": undefined,
        }
      }
    >
      <Text>
        Vanessa
         
        Ramos
      </Text>
      <Text
        style={
          {
            "color": "#A9A9A9",
            "marginTop": undefined,
          }
        }
      >
        2 months ago
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": undefined,
        "paddingStart": undefined,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
    >
      Dog in a forest at sunset dog in forest with sun r...
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          [
            {
              "borderRadius": undefined,
              "height": undefined,
              "marginVertical": undefined,
              "width": undefined,
            },
            undefined,
          ],
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": undefined,
        }
      }
    >
      <Text>
        #
        dog
         
      </Text>
      <Text>
        #
        animal
         
      </Text>
      <Text>
        #
        golden retriever
         
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Icon
        color="#B4B4B8"
        name="heart-outline"
        size={20}
      />
      <Text
        style={
          {
            "marginStart": undefined,
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
